compiler/noirc_frontend/src/hir/resolution/import.rs

```diff
@@ -298,13 +307,17 @@ fn resolve_name_in_module(
         };
 
         warning = warning.or_else(|| {
-            if can_reference_module_id(
-                def_maps,
-                importing_crate,
-                starting_mod,
-                current_mod_id,
-                visibility,
-            ) {
+            // If the path is plain, the first segment will always refer to
+            // something that's visible from the current module.
+            if (plain && index == 0)
+                || can_reference_module_id(
+                    def_maps,
+                    importing_crate,
+                    starting_mod,
+                    current_mod_id,
+                    visibility,
+                )
+            {
```

Comment:
Does `can_reference_module_id` return the wrong value here? I'm wondering why we're adding an extra argument.

Reply:
I think the issue is that when you do this:

```rust
use foo::bar
```

then `bar` is added as a private item to the module. Then, say `bar` is a module and you do `bar::baz()` it would error saying that `bar` is private.

The error didn't happen before because all imports were considered public.

So the rule here is: if it's the first segment and we find something, it can't be non-visible (it's either defined in the same module, or imported into it but if it's imported it's visible)

Reply:
Ah yeah, I ran into a similar issue when adding module visibility previously. This makes sense now.